monticore-generator/src/main/java/de/monticore/dstlgen/grammartransformation/DSL2TransformationLanguageVisitor.java

```diff
@@ -350,5 +335,31 @@ public void visit(ASTConcept srcNode) {
   public ASTMCGrammar getTfLang() {
     return tfLang;
   }
+  
+  protected boolean addASTClassProdInterfaces(ASTClassProd srcNode, ASTClassProd prod,
+      ProductionType productionType, boolean isLeftRecursive, boolean isEmpty) {
+    if (productionType == LIST) {
+      if (isEmpty) {
+        return false;
+      }
+      productionFactory.addInterfaces(srcNode.getSuperRuleList(), prod);
+      productionFactory.addInterfaces(srcNode.getSuperInterfaceRuleList(), prod);
+    }
+    else if (productionType == OPTIONAL || productionType == NEGATION
+        || productionType == REPLACEMENT) {
+      if (isLeftRecursive || isEmpty) {
+        return false;
+      }
+      productionFactory.addInterfaces(srcNode.getSuperRuleList(), prod);
+      productionFactory.addInterfaces(srcNode.getSuperInterfaceRuleList(), prod);
+    }
+    return true;
+  }
+  
+  protected void addASTClassProdAnnotations(ASTClassProd prod, boolean override) {
+    if (override) {
```

Comment:
```suggestion
    // Add @Override annotation to the production if needed
    if (override) {
```

(This class lacks documentation/comments, so why not start now :) )